src/backend/distributed/transaction/backend_data.c

```diff
@@ -682,6 +677,7 @@ UnSetGlobalPID(void)
 		SpinLockAcquire(&MyBackendData->mutex);
 
 		MyBackendData->globalPID = 0;
+		MyBackendData->distributedCommandOriginator = false;
```

Comment:
I'm a bit concerned with this unsetting. We make it false, which means this is a worker query.
Looks like we used to do this way too, but maybe we could convert to a bool pointer for true, false and NULL, or maybe a int with enums `ORIGINATOR_QUERY`, `WORKER_QUERY` and `UNASSIGNED`.

I'm not sure if this is an actual problem though 😄

Reply:
well, we don't even need to set this. I'd rather remove :D